sympy/functions/elementary/hyperbolic.py

```diff
@@ -1354,6 +1376,22 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
     def _eval_rewrite_as_log(self, x, **kwargs):
         return log(x + sqrt(x + 1) * sqrt(x - 1))
 
+    def _eval_rewrite_as_asinh(self, x, **kwargs):
+        return Piecewise((S.Zero, Eq(x, 1)),
+                         (sqrt(x - 1)/sqrt(1 - x)*(pi/2 +
+                          S.ImaginaryUnit*asinh(S.ImaginaryUnit*x)), True))
```

Comment:
This is `acosh(x).rewrite(asinh)`. I think it can be done without Piecewise:
```python
In [25]: 2*asinh(sqrt((x - 1)/2))
Out[25]: 
       ⎛    _______⎞
       ⎜   ╱ x   1 ⎟
2⋅asinh⎜  ╱  ─ - ─ ⎟
       ⎝╲╱   2   2 ⎠
```